src/ceph-volume/ceph_volume/util/arg_validators.py

```diff
@@ -28,36 +32,98 @@ def _format_device(self, device):
             return device.path
         return device
 
-    def _is_valid_device(self, dev_path):
-        device = Device(dev_path)
+    def _is_valid_device(self):
         error = None
-        if not device.exists:
-            error = "Unable to proceed with non-existing device: %s" % dev_path
+        if not self._device.exists:
+            error = "Unable to proceed with non-existing device: %s" % self.dev_path
         # FIXME this is not a nice API, this validator was meant to catch any
         # non-existing devices upfront, not check for gpt headers. Now this
         # needs to optionally skip checking gpt headers which is beyond
         # verifying if the device exists. The better solution would be to
         # configure this with a list of checks that can be excluded/included on
         # __init__
-        elif device.has_gpt_headers and not self.gpt_ok:
-            error = "GPT headers found, they must be removed on: %s" % dev_path
-        if device.has_partitions:
-            raise RuntimeError("Device {} has partitions.".format(dev_path))
+        elif self._device.has_gpt_headers and not self.gpt_ok:
+            error = "GPT headers found, they must be removed on: %s" % self.dev_path
+        if self._device.has_partitions:
+            raise RuntimeError("Device {} has partitions.".format(self.dev_path))
         if error:
             raise argparse.ArgumentError(None, error)
-        return device
+        return self._device
 
 
-class ValidBatchDevice(ValidDevice):
+class ValidZapDevice(ValidDevice):
+    def __call__(self, dev_path):
+        super().get_device(dev_path)
+        return self._format_device(self._is_valid_device())
+
+    def _is_valid_device(self, raise_sys_exit=True):
+        super(ValidZapDevice, self)._is_valid_device()
```

Comment:
NIT, I see all the `__call__` functions using just `super().` while all the `_is_valid_device` functions use `super(<class-name>, self).`. It seems like just `super()` is the newer, python3 way of doing it. Do we want to use that syntax across all super calls we're doing here?

Reply:
I think I introduced this inconsistency a bit out of ignorance, to be honest 😅 
Let's modify it indeed 👍